pepsico/comparison.py

```diff
@@ -1,14 +1,136 @@
 import xarray as xr
+import os
+import re
+
+
+ds_zarr = '/data/remic/mydatafiles/zarr/test/hurs'
+ds_nc_dir = '/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global/daily/historical/MPI-ESM1-2-HR/'
+
+#only using NetCDF files that contain 'hurs' variable
+def contains_hurs(filename):
+    return 'hurs' in filename
+
+#extracting years from nc file
+def extract_years(filename):
+    match = re.search(r'(\d{4})_(\d{4})\.nc$', filename)
+    if match:
+        start_year = int(match.group(1))
+        end_year = int(match.group(2))
+        return start_year, end_year
+    else:
+        raise ValueError("Unable to find years")
+
+
+#function that compares nc files(hurs) vs zarred file
+def compare_datasets(nc_file_path, ds_zarr):
+    #returns true if datasets are identical, false otherwise
+
+    #open datasets
+    zarr_dSet = xr.open_zarr(ds_zarr)
+    nc_dSet = xr.open_dataset(nc_file_path)[['hurs']]
+
+
+    #extracting start and end years from nc file
+    start_year, end_year = extract_years(os.path.basename(nc_file_path))
+    time_start = f'{start_year}-01-01'
+    time_end = f'{end_year}-12-31'
+
+    #check if datasets are empty
+    if len(nc_dSet) == 0 or len(zarr_dSet) == 0:
```

Comment:
What are you trying to check for here? In practice I think the len of a Dataset is the number of variables it contains, but I can't find that documented anywhere, so you're relying on undocumented behavior here. And I can't imagine what kind of error would result in the Dataset not having any variables.